Add prepend filters method

fink3l opened the pull request:
The pull request helps in situations when you collect part of the filters statically and some need to be collected dynamically and with a higher priority.

fink3l committed:
Add prepend filters method

theofidry commented on src/DeepCopy/DeepCopy.php:
```diff
@@ -85,11 +85,33 @@ public function copy($value)
     }
 
+    /**
+     * @param Filter  $filter
```
I wouldn't add the phpdoc. When it doesn't provide any more information than the typehints (like here), it just adds noise IMO.

fink3l replied:
i think you right, thanks

theofidry commented on src/DeepCopy/DeepCopy.php:
```diff
@@ -85,11 +85,33 @@ public function copy($value)
+    /**
+     * Add filter with high priority
+     *
+     * @param Filter  $filter
```
Same here. I think the method is self-describing enough to skip the description `Add filter with high priority`, but I don't mind leaving it either.

fink3l replied:
OK

theofidry commented on src/DeepCopy/DeepCopy.php:
```diff
@@ -85,11 +85,33 @@ public function copy($value)
+     */
+    public function prependFilter(Filter $filter, Matcher $matcher): void
+    {
+        \array_unshift($this->filters, [
```
I would rather add a `use function array_unshift;` statement at the top of the file than having this leading backslash. The leading backslash is necessary in older version of PHP as `use function` is not allowed there, but in this case we do not support those old versions. So it's more a matter of style and I find the use statement more readable

fink3l replied:
I made mistake with branches, i think i made PR to 1.8. Damn, sorry for that.
I added unit tests and fix prependFilter method for 2.x version, now it's OK.

theofidry reviewed (commented):
Thanks for the PR! There is some minor changes to do but looks good to me otherwise

fink3l committed:
Cleanup after code review

- Remove unnecessary phpdoc
- Switch leading backslash array_unshift to use function
- Fix for 2.x version
- Add unit test

theofidry commented:
Thanks
